### DISCUSSION TITLE
How do we want to style Postgres identifiers within the UI?

### DISCUSSION THREAD

**Discussion by seancolsen** [mathesar-foundation/mathesar#856] [Developer Q&A] [answered]
Example:

- User is deleting a table named `Foods that don't make me say "Yuck"`.
- We display a modal confirmation and want to use a title that shows the user they're about to delete _this_ table.
- How does the title appear to the user?

Options:

- Simple string

    - **Single quotes**:

        Delete table 'Foods that don't make me say "Yuck"'?

    - **Double quotes**:

        Delete table "Foods that don't make me say "Yuck""?

- Rich text

    - **Italics**:

        Delete table _Foods that don't make me say "Yuck"_?

    - **Bold**:

        Delete table **Foods that don't make me say "Yuck"**?

    - **Custom formatting**:

        Delete table `Foods that don't make me say "Yuck"`?

More thoughts:

- This same question applies to all identifiers i.e. tables, columns, views, etc.
- Using simple strings is quite a bit easier on the front end because there are a variety of contexts in which we need to place an identifier within a sentence such as error messages, confirmations. There are situations where if we use rich text, we either need to create a component for that specific text, or we need to use [{@html}](https://svelte.dev/docs#html) (which I really don't want to do because of the we'd need to manually sanitize the entity name to avoid XSS).

**Comment by seancolsen** [comment]
- I'd like to use rich text as much as possible.
- For the rich text, I'd like to create an `Identifier` component for this purpose. That way it's easy to adjust the styling later if needed.
- Within that component, I'd like to use italics for the time being.
- In situations where it's not easy/possible to use rich text, I'd like to use double quotes as a fallback.

**Reply by ghislaineguerin** [reply]
@seancolsen I've been thinking about this same thing when including object references in the UI. I'd go with rich text as well. I like the custom formatting option the most.

**Comment by ghislaineguerin** [comment, marked as answer]
I'd go with rich text and custom formatting. One thing to consider is how we'd handle line breaks and truncation in general.

**Reply by kgodey** [reply]
I concur with rich text and custom formatting. It's most clear that it's an identifier.

**Reply by seancolsen** [reply]
Cool, I'll go with this. In 3265edf4207b7d1c0d98ce4bac4754cf91726453 (within #857) I introduced an `Identifier` component for this purpose. I used italics for the time being, but I'll adjust it a bit, and it'll be easy to tweak as we move forward too once we start using this component.
